Added footnote subconverter.

shylux opened the pull request:
Converts marked footnotes in html blocks to latex footnotes.
The html requires a `span` with the class `footnote` and a title attribute:  
`<span class="footnote" title="footnote text">some text</span>`

The footnote is appended to the end of the span. The text in the title is the footnote text which is placed at the bottom of the page:
![screen shot 2016-02-17 at 12 21 04](https://cloud.githubusercontent.com/assets/1375745/13108043/07dfefdc-d571-11e5-90f7-a71db21861d5.png)
![screen shot 2016-02-17 at 12 21 10](https://cloud.githubusercontent.com/assets/1375745/13108045/0a2cf7c6-d571-11e5-83ab-c16ec739e7f3.png)

jone commented on ftw/pdfgenerator/html2latex/subconverters/footnote.py:
```diff
@@ -0,0 +1,25 @@
+        if 'footnote' not in self.match.group('class').split():
+            return
+
+        title = re.search(r'title="(?P<title>.*?)"', self.get_html())
```
I propose this change:
```diff
- r'title="(?P<title>.*?)"'
+ r'title="(?P<title>[^"]*)"'
```
Because we want to match exactly the `title` attribute value, which is terminated with the next `"`. It is not much of a difference because the old expression was non-greedy but I'd prefer to explicitly consume all non-quotes in the group. :wink:

jone replied:
I'd also rename the variable `title` to `title_match` since the regex-match is stored, not the actual title value as I'd have expected from the name.

jone commented:
- [ ] I think the footnote text should be converted too.
- [ ] Can you add some tests with umlauts in footnote text and wrapped text?

shylux commented:
@jone Fixed the regex, renamed some variables and changed footnote storage to `data-footnote` attribute to reflect changes in ftw.book.
It also converts the footnote text now. This is how it looks:
![screen shot 2016-02-18 at 12 29 40](https://cloud.githubusercontent.com/assets/1375745/13142211/e62f3d40-d63b-11e5-9171-b5e69d40eea2.png)

jone commented:
Awesome! :+1: :sparkles: 
- [ ] One last thing: could you add an example in the readme with the markup a footnote must have?

With wrapped-text I meant: `<span class="footnote"..>wrapped text</span>`, you already added an umlaut test for that :wink:

shylux commented:
@jone added markup example in readme
